figshare/figshare.py

```diff
@@ -290,3 +293,37 @@ def get_file_details(self, article_id, file_id):
         response = issue_request('GET', url,
                                  headers=self.get_headers(token=self.token))
         return response
+
+    def retrieve_files_from_article(self, article_id, outpath=''):
+        """ Retrieve files and save them locally.
+
+        By default, files will be stored in the current working directory
+        under a folder called figshare_<article_id> by default.
+        Specify <outpath> for: <outpath>/figshare_<article_id>
+
+        Parameters
+        ----------
+        article_id : str or int
+            Figshare article ID
+
+        """
+
+        # Handles missing trailing forward slash
+        if outpath != '':
+            if outpath[-1] != '/':
+                outpath += '/'
+
+        # Get list of files
+        file_list = self.list_files(article_id)
+
+        n_files = len(file_list)
+        if n_files != 0:
+            dir0 = outpath+"figshare_{0}/".format(article_id)
+            os.makedirs(dir0, exist_ok=True) # This might require Python >=3.2
+
+            for ii in range(n_files):
```

Comment:
why `ii` rather than the more conventional `i`. Also could you just iterate through file_list and avoid indexing altogether?

Reply:
Done on using file list.  I find `ii` easier to spot in code, and it is helpful for easy replace using basic text editor.